### PR TITLE
#536: Lint: Apply minimal error-related rules

### PR DESCRIPTION
**Pull request by fregante**
Based on / blocked by #562 https://github.com/pixiebrix/pixiebrix-extension/pull/564

Part of https://github.com/pixiebrix/pixiebrix-extension/issues/536

This PR:

- autofixes 2 very common but rules
- autofixes `unicorn/prevent-abbreviations` only for errors
- drops unnecessary `.catch(e => reportError(e))` wrapper

It's easy to review because it's just essentially one minor variable rename across 94 files.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -39,6 +39,18 @@
...
+    "unicorn/catch-error-name": "error",
+    "unicorn/prevent-abbreviations": [
+      "error",
+      {
+        "replacements": {
+          "err": {
+            "error": true
+          }
+        }
+      }
```
Comment (fregante) [inline .eslintrc:53]: This inline configuration is temporary. Once most of unicorn is added, we'll just use `plugin:security/recommended` with exceptions instead

File: src/contrib/google/auth.ts [hunk 2]
```diff
@@ -82,13 +82,16 @@ class PermissionsError extends Error {
   }
 }
 
-export async function handleRejection(token: string, err: any): Promise<Error> {
-  console.debug("Google rejected request", { err });
-  if (err.result == null) {
-    return err;
+export async function handleRejection(
+  token: string,
+  error: any
```
Comment (twschiller) [inline src/contrib/google/auth.ts:87]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/pixiebrix/pixiebrix-extension/pullRequest?prid=7635933)
```suggestion
  error: unknown
```

Comment (fregante) [reply src/contrib/google/auth.ts:87]: Heh, this is what I was talking about in https://github.com/pixiebrix/pixiebrix-extension/issues/536#issuecomment-865241440. I can't rename this variable without being pinged about another issue 😁

File: src/contrib/google/auth.ts [hunk 3]
```diff
@@ -82,13 +82,16 @@
...
```
Comment (twschiller) [inline src/contrib/google/auth.ts:87]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/pixiebrix/pixiebrix-extension/pullRequest?prid=7635933)
```suggestion
  error: unknown
```

File: .eslintrc (hunk 1 above)
Comment (twschiller) [reply .eslintrc:54]: > Maybe incrementally adding rules is what you had in mind with "fix as we go"

By "fix as we go" I meant enabling the recommended rules and enforcing them on new PRs but not cleaning up existing files until we touch them again in a PR. That's a benefit of Codacy only providing feedback on files in the PR

File: src/contrib/google/auth.ts (hunk 3 above)
Comment (twschiller) [inline src/contrib/google/auth.ts:87]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/pixiebrix/pixiebrix-extension/pullRequest?prid=7635933)
```suggestion
  error: unknown
```

Comment (twschiller) [inline src/contrib/google/auth.ts:87]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/pixiebrix/pixiebrix-extension/pullRequest?prid=7635933)
```suggestion
  error: unknown
```

**Review by twschiller — APPROVED** [review]

**Pull request merged**